service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -70,11 +77,6 @@ public static DigitalTwinAsyncClient createFromConnectionString(String connectio
      */
     public <T> Observable<T> getDigitalTwin (String digitalTwinId, Class<T> clazz)
     {
-        if(clazz == null)
```

Comment:
This logic is only needed in the max overload to make testing more efficient. I've cleaned up duplicate logic.